src/tfp_backend/Code_gen.ml

```diff
@@ -193,8 +193,11 @@ let get_params p =
 
 let pp_shapes ppf p =
   let pp_shape ppf (_, {Program.out_unconstrained_st; _}) =
-    pf ppf "(nchains__, @[<hov>%a@])" (list ~sep:comma pp_expr)
-      (SizedType.get_dims out_unconstrained_st)
+    let shapes_lst = SizedType.get_dims out_unconstrained_st in
+    if List.length shapes_lst > 0 then
```

Comment:
I think another way to write a version that casts each shape individually is to keep the original but wrap pp_expr with code to wrap it in a tf.cast call, 

```ocaml
let cast_expr ppf e = pf ppf "tf__.cast(%a, tf__.int32)" pp_expr e in
pf ppf "(nchains__, @[<hov>%a@])" (list ~sep:comma cast_expr)
      (SizedType.get_dims out_unconstrained_st)
```

(not sure if it's worth putting a format box around the cast argument, didn't seem like it).

Reply:
Thanks! That's certainly a better way to do this.